File: app/src/androidTest/java/com/star_zero/example/androidmvvm/di/MockAppModule.java
package com.star_zero.example.androidmvvm.di;

import com.star_zero.example.androidmvvm.domain.task.TaskRepository;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class MockAppModule {

    @Provides
    @Singleton
    TaskRepository provideTaskRepository() {
        return new MockTaskRepository();
    }
}

File: app/src/androidTest/kotlin/com/star_zero/example/androidmvvm/di/MockAppModule.kt
package com.star_zero.example.androidmvvm.di

import android.arch.lifecycle.ViewModelProvider
import com.star_zero.example.androidmvvm.domain.task.TaskRepository
import com.star_zero.example.androidmvvm.presentation.shared.viewmodel.ViewModelFactory
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module(subcomponents = arrayOf(ViewModelSubComponent::class))
class MockAppModule {

    @Provides
    @Singleton
    fun provideTaskRepository(): TaskRepository {
        return MockTaskRepository()
    }

    @Provides
    @Singleton
    fun provideViewModelFactory(viewModelSubComponent: ViewModelSubComponent.Builder): ViewModelProvider.Factory {
        return ViewModelFactory(viewModelSubComponent.build())
    }
}
